core/types/uint256.go

```diff
@@ -95,21 +133,55 @@ func (ua Uint256Array) Value() (driver.Value, error) {
 
 var _ driver.Valuer = (*Uint256Array)(nil)
 
-// Scan implements the sql.Scanner interface.
-func (ua *Uint256Array) Scan(src interface{}) error {
-	switch s := src.(type) {
-	case []string:
-		*ua = make(Uint256Array, len(s))
-		for i, str := range s {
-			u, err := Uint256FromString(str)
-			if err != nil {
-				return err
-			}
-
-			(*ua)[i] = u
-		}
+// Uint256Array is a slice of Scanners. pgx at least is smart enough to make
+// this work automatically!
+// Another approach is to implement pgx.ArraySetter and pgx.ArrayGetter like
+// similar in effect to:
+//   type Uint256Array pgtype.FlatArray[*Uint256]
+
+// Well, as long as Uint256Array is just an underlying []*Uint256, pgx works fine
+// without implementing these interfaces. Also, these methods have pgtype types on
+// them, so I'm commenting it for now, may delete.
+/*
+var _ pgtype.ArrayGetter = Uint256Array{}
+var _ pgtype.ArrayGetter = (*Uint256Array)(nil)
+
+func (ua Uint256Array) Dimensions() []pgtype.ArrayDimension {
+	if ua == nil {
 		return nil
 	}
+	return []pgtype.ArrayDimension{{Length: int32(len(ua)), LowerBound: 1}}
+}
+
+func (ua Uint256Array) Index(i int) any { return ua[i] }
 
-	return fmt.Errorf("cannot convert %T to Uint256Array", src)
+func (ua Uint256Array) IndexType() any {
+	return (*Uint256)(nil) // &Uint256{}
+}
+
+func pgDimCard(dimensions []pgtype.ArrayDimension) int {
+	if len(dimensions) == 0 {
+		return 0
+	}
+	elementCount := int(dimensions[0].Length)
+	for _, d := range dimensions[1:] {
+		elementCount *= int(d.Length)
+	}
+	return elementCount
 }
+
+var _ pgtype.ArraySetter = (*Uint256Array)(nil)
+
+func (ua *Uint256Array) SetDimensions(dimensions []pgtype.ArrayDimension) error {
+	if dimensions == nil {
+		*ua = nil
+		return nil
+	}
+	*ua = make(Uint256Array, pgDimCard(dimensions))
+	return nil
+}
+
+func (ua Uint256Array) ScanIndex(i int) any { return &ua[i] }
+
```

Comment:
are you still planning / debating to use this commented out code?

Reply:
Cleaned it out.  PR squashed.